frontend/apps/reader/modules/readertypeset.lua

```diff
@@ -56,63 +59,69 @@ function ReaderTypeset:onReadSettings(config)
+                   or G_reader_settings:readSetting("copt_render_dpi")
+                   or 96
     self:setRenderDPI(self.render_dpi)
 
     -- uncomment if we want font size to follow DPI changes
     -- self.ui.document:setRenderScaleFontWithDPI(1)
 
     -- set page margins
-    local h_margins = config:readSetting("copt_h_page_margins") or
-        G_reader_settings:readSetting("copt_h_page_margins") or
-        DCREREADER_CONFIG_H_MARGIN_SIZES_MEDIUM
-    local t_margin = config:readSetting("copt_t_page_margin") or
-        G_reader_settings:readSetting("copt_t_page_margin") or
-        DCREREADER_CONFIG_T_MARGIN_SIZES_LARGE
-    local b_margin = config:readSetting("copt_b_page_margin") or
-        G_reader_settings:readSetting("copt_b_page_margin") or
-        DCREREADER_CONFIG_B_MARGIN_SIZES_LARGE
+    local h_margins = config:readSetting("copt_h_page_margins")
+                   or G_reader_settings:readSetting("copt_h_page_margins")
+                   or DCREREADER_CONFIG_H_MARGIN_SIZES_MEDIUM
+    local t_margin = config:readSetting("copt_t_page_margin")
+                  or G_reader_settings:readSetting("copt_t_page_margin")
+                  or DCREREADER_CONFIG_T_MARGIN_SIZES_LARGE
+    local b_margin = config:readSetting("copt_b_page_margin")
+                  or G_reader_settings:readSetting("copt_b_page_margin")
+                  or DCREREADER_CONFIG_B_MARGIN_SIZES_LARGE
     self.unscaled_margins = { h_margins[1], t_margin, h_margins[2], b_margin }
     self:onSetPageMargins(self.unscaled_margins)
-    self.sync_t_b_page_margins = config:readSetting("copt_sync_t_b_page_margins") or
-        G_reader_settings:readSetting("copt_sync_t_b_page_margins") or 0
+    self.sync_t_b_page_margins = config:readSetting("copt_sync_t_b_page_margins")
+                              or G_reader_settings:readSetting("copt_sync_t_b_page_margins")
+                              or 0
     self.sync_t_b_page_margins = self.sync_t_b_page_margins == 1 and true or false
 
     -- default to disable TXT formatting as it does more harm than good
-    self.txt_preformatted = config:readSetting("txt_preformatted") or
-        G_reader_settings:readSetting("txt_preformatted") or 1
+    self.txt_preformatted = config:readSetting("txt_preformatted")
+                         or G_reader_settings:readSetting("txt_preformatted")
+                         or 1
     self:toggleTxtPreFormatted(self.txt_preformatted)
 
     -- default to disable smooth scaling for now.
-    self.smooth_scaling = config:readSetting("smooth_scaling")
-    if self.smooth_scaling == nil then
+    if config:has("smooth_scaling") then
+        self.smooth_scaling = config:isTrue("smooth_scaling")
+    else
         local global = G_reader_settings:readSetting("copt_smooth_scaling")
-        self.smooth_scaling = (global == nil or global == 0) and 0 or 1
+        self.smooth_scaling = (global == nil or global == 0) and false or true
```

Comment:
(This and the next one were settings you added - so I guess you've checked it's fine for credocument and cre.cpp to use false/true instead of 0/1 - in some places for some other settings it wasn't :)

Reply:
Yeah, that one was actually arguably a bug I sneakily fixed ^^.

(It worked because for all intents and purposes when doing boolean checks 0 is false and 1 is true), but it made the whole thing weird given the legacy settings above that *do* expect a 0/1 ;).